packages/react-instantsearch-hooks/src/lib/useSearchResults.ts

```diff
@@ -21,10 +21,15 @@ export function useSearchResults(): SearchResultsApi {
 
   useEffect(() => {
     function handleRender() {
-      setSearchResults({
-        results: searchIndex.getResults()!, // Results can't be `null` after the first render.
-        scopedResults: searchIndex.getScopedResults(),
-      });
+      const results = searchIndex.getResults();
+
+      // results can be null when first search is stalled, in this case we skip the update
+      if (results !== null) {
```

Comment:
I hesitated to also check if results are different from the previous, which could be the case when a subsequent search is stalled, which currently would trigger a rerender for nothing.
However, `useSearchResults` is only used in `useInstantSearch` which would anyway trigger rerender even if `useSearchResults` did not change, so I opted to keep the code simpler.

Reply:
Why would `useInstantSearch` re-render anyway even if the data returned from `useSearchResults` didn't change?

Reply:
Because it [also uses `useSearchState`](https://github.com/algolia/react-instantsearch/blob/a416fc9ec2d28ccc4ea039cbad3cb2c9185afe77/packages/react-instantsearch-hooks/src/hooks/useInstantSearch.ts#L21-L22) which [sets state on every `render` event](https://github.com/algolia/react-instantsearch/blob/a416fc9ec2d28ccc4ea039cbad3cb2c9185afe77/packages/react-instantsearch-hooks/src/lib/useSearchState.ts#L56-L60)